system76driver/actions.py

```diff
@@ -1424,15 +1424,47 @@ def describe(self):
         return _('Enable ForceFullCompositionPipeline in the NVIDIA driver')
 
     def __init__(self, etcdir='/etc'):
-        self.filename = path.join(etcdir, 'profile.d', 's76-nvidia-fullcomp.sh')
+        self.oldfilename = path.join(etcdir, 'profile.d', 's76-nvidia-fullcomp.sh')
+        self.filename = path.join(etcdir, 'xprofile')
+    
+    def read(self):
+        return open(self.filename, 'r').read()
 
     def get_isneeded(self):
-        return not os.path.exists(self.filename)
+        if os.path.exists(self.oldfilename):
+            return True
+        elif not os.path.exists(self.filename):
+            return True
+        elif not "ForceFullCompositionPipeline" in self.read():
+            return True
+        else:
+            return False
 
     def perform(self):
-        content = '# Added by system76-driver.\n'
-        content += '# Force a full composition pipeline to prevent stuttering.\n'
-        content += 'nvidia-settings --assign CurrentMetaMode="nvidia-auto-select +0+0 { ForceFullCompositionPipeline = On }"\n'
+        if os.path.exists(self.oldfilename):
+            try:
+                os.remove(self.oldfilename)
+            except:
+                pass
+        if os.path.exists(self.filename):
+            if "ForceFullCompositionPipeline" in self.read():
+                return
+            content = self.read_and_backup()
+            content += '\n# Added by system76-driver.\n'
+        else:
+            content = '# Added by system76-driver.\n'
+        content += '# Force a full composition pipeline to prevent stuttering.\n\n'
+        content += '# Get the current display settings.\n'
+        content += 'oldmode=$(echo $(nvidia-settings -q CurrentMetaMode) | sed -e \'s/.*:: //g\')\n\n'
+        content += '# If there were no old settings, then apply a default setting;\n'
+        content += '# otherwise, add ForceFullCompositionPipeline to the old settings.\n'
+        content += 'if [ -z "$oldmode" ]\n'
+        content += 'then\n'
+        content += '    newmode="nvidia-auto-select +0+0 { ForceFullCompositionPipeline = On}"\n'
+        content += 'else\n'
+        content += '    newmode=$(echo $oldmode | sed \'s/}/, ForceFullCompositionPipeline=On}/g\')\nfi\n\n'
```

Comment:
```
newmode=$(echo $oldmode | sed 's/}/, ForceFullCompositionPipeline=On}/g')
          ^-- SC2001: See if you can use ${variable//search/replace} instead.
               ^-- SC2086: Double quote to prevent globbing and word splitting.
```

```suggestion
        content += '    newmode=$(echo "$oldmode" | sed \'s/}/, ForceFullCompositionPipeline=On}/g\')\nfi\n\n'
```

Reply:
I've added the double-quotes, but I'm keeping `sed` instead of the Bash-specific find/replace suggested by `SC2086` because we definitely have users who change their shells.